core/src/main/scala/cr/pulsar/Reader.scala

```diff
@@ -71,6 +73,22 @@ object Reader {
                 }
               }
             )
+
+          override def read1: F[Option[Message[E]]] =
+            F.delay(c.hasMessageAvailableAsync).futureLift.map(Boolean.unbox).flatMap {
+              case false => F.pure(None)
+              case true =>
+                F.delay(c.readNextAsync()).futureLift.flatMap { m =>
```

Comment:
You can have this whole construct assigned to a variable and then use `whenA` instead of pattern matching on the Boolean.

Reply:
Nevermind, `whenA` only works for `F[Unit]`...